import-automation/progress-dashboard-rest/test/test_system_run_list.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for system_run_list.py.
+"""
+
+import unittest
+from unittest import mock
+
+from app.model import system_run_model
+from app.resource import system_run_list
+from test import utils
+
+_MODEL = system_run_model.SystemRunModel
+
+
+class SystemRunListTest(unittest.TestCase):
+    """Tests for SystemRunList."""
+
+    # TODO(intrepiditee): Replace with module level setup
+    @classmethod
+    def setUpClass(cls):
+        cls.emulator = utils.start_emulator()
+
+    @classmethod
+    def tearDownClass(cls):
+        utils.terminate_emulator(cls.emulator)
+
+    @mock.patch('app.utils.create_datastore_client',
+                utils.create_test_datastore_client)
+    def setUp(self):
+        """Injects several system runs to the database before every test."""
+        self.resource = system_run_list.SystemRunList()
+        runs = [
+            {_MODEL.repo_name: 'dddd', _MODEL.pr_number: 0},
+            {_MODEL.repo_name: 'data', _MODEL.pr_number: 1},
```

Comment:
is this to test that duplicates are handled a given way?

Reply:
They are not duplicates. Here it is just adding four runs to the database